009.md

```diff
@@ -0,0 +1,197 @@
+
+The most trivial tangle is the classic SSB feed.  In this case, the tangle for
+feed `A` is defined as:
+
+1. **Candidates messages**:
+   - MUST have `author` field equal to `A`
+2. **Recipe**:
+   1. Find the message with `sequence` field equal to `1` as your "root"
+       - Check it's signed correctly by `A`
+   2. Find the message with `sequence` field equal to `i` for some `i` > `1`
+       - Check it's signed correctly by `A`
+       - Check its `previous` field points to a message with `sequence` field
+         equal to `i-1`
+
+
+```mermaid
+flowchart RL
+A(A):::root;B(B);C(C);D(D);
+
+subgraph "feed"
+  direction RL
+D-->C-->B-->A
+end
+
+classDef default fill: #e953da, stroke-width: 0, color: white
+classDef root fill: #4c44cf, stroke-width: 0, color: white
+classDef cluster stroke:#e953da,fill:none;
+```
+
+The tangle data these messages carry looks like:
+```javascript
+a => { sequence: 1, previous: null, ... }
+b => { sequence: 2, previous: a, ... }
+c => { sequence: 3, previous: b, ... }
+d => { sequence: 4, previous: c, ... }
+```
+
+(The fields `author`, `signature`, `timestamp`, `hash`, and `content` have been
+ommited here to make the backlinking pattern clearer)
+
+This is a special case of a tangle, for a few reasons:
+
+- This tangle does not have `m.content.tangles[x]` with `root` and `previous`
+- The DAG is a single chain, with no branches or merges
+
+
+### Multi-author tangle
+
+While the classic feed is a special case of a tangle without branches, in the
+case of multiple authors, we need to support branches and merges. This is
+because different authors may contribute concurrently, or while offline.
+
+```mermaid
+flowchart RL
+A(A):::root;B(B);X(X);Y(Y);M(M);
+
+M-->X-->B-->A
+M-->Y--->B
+
+classDef default fill: #e953da, stroke-width: 0, color: white
+classDef root fill: #4c44cf, stroke-width: 0, color: white
+```
+_Diagram where messages X, Y were both published concurrently (so were unaware
+of one another). Message M is aware of both X and Y, and extends the tangle from
+them.  M is now the new "tip" of the tangle._
+
+The tangle data these messages carry looks like:
+
+```javascript
+A => { root: null, previous: null }
+B => { root: A,    previous: [A] }
+X => { root: A,    previous: [B] }
+Y => { root: A,    previous: [B] }
+M => { root: A,    previous: [X, Y] }
+```
+
+Where:
+- `root` is the ID of the root message of the tangle
+- `previous` is an Array of message IDs of the tip(s) of the tangle at the time
+  this message was published
+
+Notes:
+- the root message cannot include its own ID (not known until published), so it
+  sets its `root` value as `null`, which means "I am a root". 
+- the "tangle data" `{ root, previous }` for e.g. a "chess tangle" SHOULD be
+  stored at `msg.value.content.tangles.chess`.
+- some tangles may have a _recipe_ that define only some messages as valid e.g.
+  only messages authored by peers the original author follows.
+
+
+## Considerations
+
+TODO
```

Comment:
```suggestion
**Security Considerations** - NONE

**Privacy Considerations** - if a tangle is encrypted (e.g. to a group) then all messages in it should be encrypted similarly, because message backlinks reveal the other authors who may have participated in the tangle, which may be a privacy risk. This means that you also want to be careful of revealing individual messages in a tangle (a feature box1, box2 support)

**Compatibility Considerations** - it's important that when defining a tangle that you the candidate messages and recipe are very clearly defined. The recipe may also need to include how a tangle should be presented to users (e.g. whether a thread tangle is linearised, and how)
```

Reply:
Are there security considerations? Potential problems with privacy? (Legit asking, I don't know if there are)

Or, are there problems with backwards compatibility? Maybe something about compatibility with classic threads (root, branch, fork)?

Reply:
Great additions. I didn't know about that privacy consideration, and now that it's written down, it is pretty important!